mlflow/telemetry/schemas.py

```diff
@@ -0,0 +1,68 @@
+import platform
+import sys
+import uuid
+from enum import Enum
+from typing import Optional
+
+import pydantic
+from pydantic import Field
+
+from mlflow.version import VERSION
+
+
+class APIStatus(str, Enum):
+    UNKNOWN = "unknown"
+    SUCCESS = "success"
+    FAILURE = "failure"
+
+
+class ModelType(str, Enum):
+    MODEL_PATH = "model_path"
+    MODEL_OBJECT = "model_object"
+    PYTHON_MODEL = "python_model"
+    CHAT_MODEL = "chat_model"
+    CHAT_AGENT = "chat_agent"
+    RESPONSES_AGENT = "responses_agent"
+
+
+class LogModelParams(pydantic.BaseModel):
+    flavor: str
+    model: ModelType
+    pip_requirements: bool = False
+    extra_pip_requirements: bool = False
+    code_paths: bool = False
+    params: bool = False
+    metadata: bool = False
+
+    model_config = {"use_enum_values": True}
+
+
+class AutologParams(pydantic.BaseModel):
+    flavor: str
+    disable: bool
+    log_traces: bool
+    log_models: bool
+
+
+class GenaiEvaluateParams(pydantic.BaseModel):
+    scorers: list[str]
+    predict_fn: bool = False
+
```

Comment:
do we really need these dataclasses? Can we use `dict` or `TypedDict` to avoid unnecessary data conversion?

Reply:
Yea TypedDict sounds good

Reply:
Why do we need default values?

Reply:
It's not necessary, but if it has default values then we don't need to set it when parsing the params.
Anyways, I think TypedDict is fine as well, let me update